src/Analyzers/CSharp/Analyzers/MakeStructMemberReadOnly/CSharpMakeStructMemberReadOnlyAnalyzer.cs

```diff
@@ -167,6 +167,10 @@ private static bool IsPotentiallyValueType(IOperation? instance)
         // Now walk up the instance-operation and see if any operation actually or potentially mutates this value.
         for (var operation = instanceOperation.Parent; operation != null; operation = operation.Parent)
         {
+            // Had a parent we didn't understand.  Assume that 'this' could be mutated.
+            if (operation.Kind == OperationKind.None)
+                return true;
```

Comment:
💭 Makes sense as a fix, but if in the future we add IOperation support for some feature it's unfortunate that it could lead to new false positives in the analyzer (not for _this_ case because we have a test for it, but for other unknown cases).